File: digitalmaktabapi/Models/Class.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace digitalmaktabapi.Models
{
    [Table("Class")]
    public class Class : Base
    {
        public required Guid SchoolId { get; set; }
        public required School School { get; set; }
        public required ClassName ClassName { get; set; }
        public required Guid BranchId { get; set; }
        public required Branch Branch { get; set; }
        public required Guid CalendarYearId { get; set; }
        public required CalendarYear CalendarYear { get; set; }
        public required ClassType ClassType { get; set; }
        public required Shift Shift { get; set; }
        public required Guid TeacherId { get; set; }
        public required Teacher Teacher { get; set; }
        public required ICollection<ClassSubject> ClassSubjects { get; set; } = [];
        public required ICollection<Attendance> Attendances { get; set; } = [];
    }
}

File: digitalmaktabspa/src/models/Class.ts
import { Base } from "./Base";
import { Branch } from "./Branch";
import { CalendarYear } from "./CalendarYear";
import { ClassName } from "./ClassName";
import { ClassType } from "./ClassType";
import { Enrollment } from "./Enrollment";
import { School } from "./School";
import { Shift } from "./Shift";
import { Teacher } from "./Teacher";

export interface Class extends Base {
  school: School;
  className: ClassName;
  classNameValue: string;
  branchId: string;
  branch: Branch;
  classTypeValue: string;
  classType: ClassType;
  shift: Shift;
  shiftValue: string;
  teacher: Teacher;
  calendarYear: CalendarYear;
  enrollments: Enrollment[];
}
